Agda.cabal

```diff
@@ -247,6 +246,9 @@ common language
 
   if impl(ghc >= 8.10)
     ghc-options:
+      -Werror
+         -- #6623: Cannot have -Werror on GHC < 8.10 because of (nameless!?) warning.
+         -- "Pattern match checker exceeded (2000000) iterations in a case alternative."
```

Comment:
Did you try to turn off the warning [`incomplete-patterns`](https://downloads.haskell.org/ghc/8.8.4/docs/html/users_guide/using-warnings.html#ghc-flag--Wincomplete-patterns) (and perhaps also [`incomplete-uni-patterns`](https://downloads.haskell.org/ghc/8.8.4/docs/html/users_guide/using-warnings.html#ghc-flag--Wincomplete-uni-patterns)) for old versions of GHC? I think that is better than shipping code that generates a warning when compiled.

Reply:
I tried, adding `-Wno-warn-incomplete-patterns`, but it didn't have an effect on this warning.  I also noticed that this warning does not come with a name (unlike all other warnings), so maybe it cannot be turned off?

Reply:
Someone else [had](https://gitlab.haskell.org/ghc/ghc/-/issues/13464) a similar problem, and the solution was to disable `overlapping-patterns` as well.

Reply:
0dd48ba0a93f5fd06195534cc4e2aa6cd4523ab1